pulp_2to3_migration/app/migration.py

```diff
@@ -320,6 +321,67 @@ def create_repo_version(progress_rv, pulp2_repo, pulp3_remote=None):
         pulp2_repo(Pulp2Repository): a pre-migrated repository to create a repo version for
         pulp3_remote(remote): a pulp3 remote
     """
+    def detect_path_overlap(paths):
+        """
+        Check for valid POSIX paths (ie ones that aren't duplicated and don't overlap).
+
+        Overlapping paths are where one path terminates inside another (e.g. a/b and a/b/c).
+
+        NOTE: The logic is copied from pulpcore.app.files.validate_file_paths().
```

Comment:
Using directly from pulpcore is painful because I would need to extract paths from the exception messages.